integration_tests/symbolics_02.py

```diff
@@ -1,35 +1,51 @@
-from sympy import Symbol, pi
+from sympy import Symbol, pi, Add, Mul, Pow
 from lpython import S
 
 def test_symbolic_operations():
     x: S = Symbol('x')
     y: S = Symbol('y')
-    p1: S = pi
-    p2: S = pi
+    pi1: S = pi
+    pi2: S = pi
     
     # Addition
     z: S = x + y
+    z1: bool = z.func == Add
+    z2: bool = z.func == Mul
     assert(z == x + y)
+    assert(z1 == True)
+    assert(z2 == False)
```

Comment:
In addition to this, let's also add tests like this:
```python
if z.func == Add:
    assert True
else:
    assert False
```
as well as:
```python
assert z.func == Add
```
To ensure that the `z.func == Add` idiom works in all the situations:
* assigning to a bool variable (already tested)
* in `if`
* in `assert`

Reply:
This should still be addressed.

Reply:
Can you add a test `assert z.func == Add` ?  I don't see it, unless I missed it.

Reply:
Will remember this and add it in a subsequent pr